src/lib/notes.ts

```diff
@@ -1,23 +1,213 @@
+    const noteFromCue = {
+      text: context?.cueFormElemRef?.current?.input.value,
+      startTime: context.currentTime,
+      endTime: context.currentTime,
+    } as VTTCue
+
+    const {lesson: resource}: any = context.resource as LessonResource
+
+    await axios
+      .post(`/api/lessons/notes/${resource.slug}`, {
+        text: noteFromCue.text,
+        startTime: context.currentTime,
+        endTime:
+          context.currentTime + readingTime(noteFromCue.text).time / 1000,
+        state: context.writingCueNoteVisibility,
+        contact_id: context.viewer.contact_id,
+        image: context.viewer.avatar_url,
+      })
+      .catch(() => {
+        send('FAIL')
+      })
+      .then(({data}: any) => {
+        // pass cue text with props to get an id right away
+        // from response and avoid making another request
+        const {start_time: startTime, end_time: endTime, ...noteData} = data
+        const cueNoteText = JSON.stringify({startTime, endTime, ...noteData})
+        const cue = new VTTCue(
+          noteFromCue.startTime,
+          noteFromCue.endTime,
+          cueNoteText,
+        )
+        send({type: 'DONE_SUBMITTING_NOTE', cue: cue})
+
+        // reset form
+        context.cueFormElemRef?.current?.input.blur()
+        context.cueFormElemRef?.current?.reset()
+      })
+  }
+
+export const deleteCueNote =
+  (context: VideoStateContext, _event: VideoEvent) => async (send: any) => {
+    const activeCues = context.activeCues
+    const currentCue: any = last(activeCues)
+    const cueId = JSON.parse(currentCue.text).id
+
+    await axios
+      .delete(`/api/lessons/notes/${context.resource.slug}?id=${cueId}`)
+      .catch((e) => {
+        console.log(`Failed to delete cue note: ${e.message}`)
+      })
+      .then(() => {
+        send({type: 'CLEAR_CUES'})
+      })
+  }
+
+export const loadStaffNotesForResource = async (staffNotesUrl: any) => {
+  const data = await (await fetch(staffNotesUrl)).text()
+  const notes = await parseMdxNotesFile(data)
+  return notes
+}
+
 export const loadUserNotesForResource = async (
-  contactId: string,
   lessonSlug: string,
+  viewer: Viewer,
 ) => {
+  const {contact_id, avatar_url} = viewer
   // all notes for the specific user
   // all public notes
   if (supabase) {
     const {data, error} = await supabase
       .from(process.env.RESOURCE_NOTES_TABLE_NAME || 'resource_notes')
       .select()
       .eq('resource_id', lessonSlug)
-      .or(`state.eq.published${contactId ? `,user_id.eq.${contactId}` : ``}`)
-    return {data, error}
+      .or(`state.eq.published${contact_id ? `,user_id.eq.${contact_id}` : ``}`)
+
+    // load and add user avatars to display along with notes
+    const token = getAccessTokenFromCookie()
+    const userIds = data?.map((note) => note.user_id)
+    // TODO: fix the GraphQL query in loadContactAvatars as it's not working
+    const avatars = userIds && (await loadContactAvatars(userIds, token))
```

Comment:
we can fetch user avatars from `contact_id` instead of hardcoding them in supabase. the issue here is that `contact_avatars` graphql query doesn't seem to be working.